### PR TITLE
Fix missing `setuptools` dep

### PR DESCRIPTION
**Pull request by joeyorlando**
# What this PR does

_tldr;_ I think we should install `setuptools` into our engine `Dockerfile` + in our CI env because Python 3.12 no longer installs `distutils` by default. This should unblock us from being able to merge #4656 and #4555.

**More details**

I would like to be able to merge #4656 and #4555. _However_, in both of these PRs `setuptools` is being removed from `requirements-dev.txt` ([here](https://github.com/grafana/oncall/pull/4555/files#diff-d8146d0816a943b0fa69a20399d7bbdb58e1c84c8b7933b2ba6dea7c10c410f5L113-L116) and [here](https://github.com/grafana/oncall/pull/4656/files#diff-d8146d0816a943b0fa69a20399d7bbdb58e1c84c8b7933b2ba6dea7c10c410f5L113-L116)). This leads to things breaking (CI job [#1](https://github.com/grafana/oncall/actions/runs/9865348392/job/27242117474?pr=4555#step:5:98) and [#2](https://github.com/grafana/oncall/actions/runs/10078898966/job/27864920455?pr=4656#step:5:100)) because of:
```bash
File "/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/site-packages/polymorphic/__init__.py", line 9, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

(☝️ from the `pkg_resources` [docs](https://setuptools.pypa.io/en/latest/pkg_resources.html).. "The pkg_resources module distributed with setuptools provides an API for Python libraries")

Python 3.12 made a change to no longer pre-install `distutils` ([relevant release notes](https://docs.python.org/3/whatsnew/3.12.html#:~:text=The%20third%2Dparty%20Setuptools%20package%20continues%20to%20provide%20distutils%2C%20if%20you%20still%20require%20it%20in%20Python%203.12%20and%20beyond)):
> [PEP 632](https://peps.python.org/pep-0632/): Remove the distutils package. See [the migration guide](https://peps.python.org/pep-0632/#migration-advice) for advice replacing the APIs it provided. The third-party [Setuptools](https://setuptools.pypa.io/en/latest/deprecated/distutils-legacy.html) package continues to provide distutils, if you still require it in Python 3.12 and beyond.
>
> [gh-95299](https://github.com/python/cpython/issues/95299): Do not pre-install setuptools in virtual environments created with [venv](https://docs.python.org/3/library/venv.html#module-venv). This means that distutils, setuptools, pkg_resources, and easy_install will no longer available by default; to access these run pip install setuptools in the [activated](https://docs.python.org/3/library/venv.html#venv-explanation) virtual environment.

Additionally, `setuptools` is in `pip-tools` `UNSAFE_PACKAGES` list ([related GitHub issue](https://github.com/pypa/pipenv/issues/1417#issuecomment-364795745)), hence why I think Dependabot is removing it in #4656 and #4555.

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: engine/requirements.in [hunk 1]
```diff
@@ -53,7 +53,7 @@ python-telegram-bot==13.13
 recurring-ical-events==2.1.0
 redis==5.0.1
 regex==2021.11.2
-requests==2.32.0
+requests==2.32.3
```
Comment (joeyorlando) [inline engine/requirements.in:56]: also, bump `requests` because of this message I was seeing when running `uv pip compile`:

```bash
warning: `requests==2.32.0` is yanked (reason: "Yanked due to conflicts with CVE-2024-35195 mitigation").
```

File: engine/requirements.txt [hunk 2]
```diff
@@ -34,7 +34,7 @@ cachetools==4.2.2
     # via
     #   google-auth
     #   python-telegram-bot
-celery[redis]==5.3.1
```
Comment (joeyorlando) [inline engine/requirements.txt:37]: I don't quite understand why it's dropping the `[redis]` portion here 🤔

File: engine/requirements.txt [hunk 3]
```diff
@@ -157,7 +157,7 @@ firebase-admin==5.4.0
     # via fcm-django
 flask==3.0.2
     # via slack-export-viewer
-google-api-core[grpc]==2.17.0
```
Comment (joeyorlando) [inline engine/requirements.txt:160]: similar to [this comment](https://github.com/grafana/oncall/pull/4799/files#r1711979686)

File: engine/requirements.txt [hunk 4]
```diff
@@ -416,6 +416,10 @@ rsa==4.9
     # via google-auth
 s3transfer==0.10.0
     # via boto3
+setuptools==72.1.0
+    # via
+    #   apscheduler
+    #   opentelemetry-instrumentation
```
Comment (joeyorlando) [inline engine/requirements.txt:422]: I guess there's not much we can do here? I assume this means that `apscheduler` and `opentelemetry-instrumentation` both have a dependency on `setuptools`?

File: engine/requirements.txt (hunk 2 above)
Comment (matiasb) [reply engine/requirements.txt:37]: FYI, https://github.com/astral-sh/uv/issues/1595 (I don't have a strong opinion to preserve the extra, though)

File: engine/Dockerfile [hunk 5]
```diff
@@ -27,7 +27,7 @@ RUN if [ "$TARGETPLATFORM" = "linux/arm64" ]; then \
     && rm grpcio-1.64.1-cp312-cp312-linux_aarch64.whl; \
     fi
 
-RUN pip install uv
+RUN pip install uv setuptools
```
Comment (matiasb) [inline engine/Dockerfile:30]: if setuptools is being added through requirements, does it need to be here too? (same above)

**Review by matiasb — APPROVED** [review]

File: engine/requirements.txt (hunk 2 above)
Comment (joeyorlando) [reply engine/requirements.txt:37]: thanks for the reference 👍 (seems like including it/excluding it doesn't make a big difference)

File: engine/Dockerfile (hunk 5 above)
Comment (joeyorlando) [reply engine/Dockerfile:30]: I guess _technically_ no, but I think `setuptools` is only being included in `requirements.txt`/`requirements-dev.txt` because of two transitive dependencies. If those dependencies drop their requirement on `setuptools` we'll need this anyways 🤔 

Python 3.12's release notes [mention](https://docs.python.org/3/whatsnew/3.12.html#:~:text=The%20third%2Dparty%20Setuptools%20package%20continues%20to%20provide%20distutils%2C%20if%20you%20still%20require%20it%20in%20Python%203.12%20and%20beyond):
> https://github.com/python/cpython/issues/95299: Do not pre-install setuptools in virtual environments created with [venv](https://docs.python.org/3/library/venv.html#module-venv). This means that distutils, setuptools, pkg_resources, and easy_install will no longer available by default; to access these run pip install setuptools in the [activated](https://docs.python.org/3/library/venv.html#venv-explanation) virtual environment.
